Libs/MRML/Logic/vtkMRMLApplicationLogic.h

```diff
@@ -235,6 +235,21 @@ class VTK_MRML_LOGIC_EXPORT vtkMRMLApplicationLogic
   /// Requests the application to show user interface for editing a node.
   virtual void EditNode(vtkMRMLNode* node);
 
+  /// Registers a module with its corresponding logic
+  /// \param moduleName name of the module.
+  /// \param moduleLogic pointer to logic to be associated to the module.
+  void RegisterModuleLogic(const char* moduleName, vtkMRMLAbstractLogic* moduleLogic);
```

Comment:
For consistency, I think we should change `RegisterModuleLogic/UnregisterModuleLogic` to `SetModuleLogic`

Since we are associating an already instantiated logic with the application logic and not asking the application to instantiate a module logic for us, this would be more consistent with the current use of register/unregister across the application .

Reply:
cc: @lassoan

Reply:
Agreed. This is a simple cache, so I would just use Add/Remove/Get or Set/Get (remove by setting nullptr). Maybe Set/Get is simpler.

Reply:
@lassoan, @jcfr I would prefer to use Add/Remove/Get methods. Set/Get is simpler, but it is also more unclear an less regular (depending on the parameter you pass, you get the opposite behavior out of the function). Also, during module load, there will be modules that has no logic and will trigger runtime warnings ("SetModuleLogic: module is not registered"). 

A related question here would be: do we accept setting of modules with no logic (`nullptr`)?

Reply:
All these naming options could work, we just need to choose a one that is the most consistent with already established naming conventions.

I've checked what methods are now used in MRML for accessing maps (see details below):
- Get/Set/Remove 4x (in one case "Unset" is used instead of "Remove")
- Get/Add/Remove 2x

Add/Remove is tend to be used where "Add" means that a new item will be present (we do not want to overwrite an existing item, that would be an error). For example AddSegment clearly communicates the expectation that the new segment will be added and will not overwrite any existing segments. In contrast, "Set" effectively means "AddOrUpdate", i.e., if there is an existing key then we want to overwrite that.

Since in this case we want the "AddOrUpdate" behavior (e.g., when we reload a scripted module then we'll want to update the logic) and Get/Set/Remove is used more overall anyway, **we should use Get/Set/Remove**.

To be consistent with most other Get/Set/Remove implementations, Set(..., nullptr) should be equivalent with Remove.

---

vtkMRMLNode:
  GetAttribute
  SetAttribute
  RemoveAttribute

vtkMRMLScriptedModuleNode:
  void SetParameter(const std::string& name, const std::string& value);
  void UnsetParameter(const std::string& name);
  std::string GetParameter(const std::string& name) const;

vtkMRMLSegmentationDisplayNode:
  bool GetSegmentDisplayProperties(std::string segmentID, SegmentDisplayProperties &properties);
  void SetSegmentDisplayProperties(std::string segmentID, SegmentDisplayProperties &properties);
  void SetSegmentDisplayPropertiesToDefault(const std::string& segmentId);
  void RemoveSegmentDisplayProperties(std::string segmentID);

vtkSegment:
  vtkDataObject* GetRepresentation(std::string name);
  bool AddRepresentation(std::string type, vtkDataObject* representation);
  bool RemoveRepresentation(std::string name);

vtkSegmentation:
  bool AddSegment(vtkSegment* segment, std::string segmentId = "", std::string insertBeforeSegmentId = ""); 
  void RemoveSegment(std::string segmentId);
  vtkSegment* GetSegment(std::string segmentId); 
vtkSegment:

vtkSegmentation:
  void SetTag(std::string tag, std::string value);
  void RemoveTag(std::string tag);
  bool GetTag(std::string tag, std::string &value);